### DISCUSSION TITLE
What user permission is used to run a build.

### DISCUSSION THREAD

**Discussion by tommy-mcgee** [buildbot/buildbot#6806] [General]
It is  an odd questions but let me try and explain.

I build ISO files using debian live-build tools if I run directly from terminal
**my command to build iso**  when the ISO is completed I can boot up log in no issue access to files is ok

I  can do  that as  sudo, or any specified user.

Now... if I build the ISO using buildbot regardless of sudo or any user the result is as follows 
Using the same type of shell command  as above in the factory....the ISO completes I can login  but there are permission issues for folder basically everything is read only.

that poses a question for me...what user is used when a job runs, is it the user that hosts buildbot?

 if you run the scripts directly from terminal or cron job it... the permissions are fine.
but if buildbot runs the script the permissions are read only.... not expecting a solution LOL but just thought its worth asking
Non the less thanks and great work!

**Comment by tommy-mcgee** [comment]
Just to add context right....so this happens when files are copied during a  chroot process. 
For example,  if in terminal I run 
```sudo lb build``
The end result is a completed process with permissions set as such on an example file
**-rw-r--r-- 1 root root  1882 Mar  1 11:14 peppermint-hub.png**

But If I run the same command  with the ShellCommand step using buildbot
The result is this.
**-rw------- 1 root root  1882 Mar  1 07:41 peppermint-hub.png**

I wonder what is different with buildbot's process

**Reply by seankelly** [reply]
[Check the `umask` setting in the Worker setup](https://docs.buildbot.net/current/manual/installation/worker.html#cmdoption-buildbot-worker-create-worker-umask).

**Comment by tommy-mcgee** [comment]
Thanks! I will give that a shot... I did end up with a workaround, that entails, having the worker run a shell script that SSH's in to my ISO build machine, all my permissions issues were resolved. But reading about the umask that may be the way to go.
